cli/upgrade.rs

```diff
@@ -28,14 +28,27 @@ use std::string::String;
 use tempfile::TempDir;
 use url::Url;
 
+const NO_ARCHIVE: &str = "No packaged releases available for your architecture";
 // TODO(ry) Auto detect target triples for the uploaded files.
-#[cfg(windows)]
+#[cfg(all(target_os = "windows", target_arch = "x86_64"))]
 const ARCHIVE_NAME: &str = "deno-x86_64-pc-windows-msvc.zip";
-#[cfg(target_os = "macos")]
+#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
 const ARCHIVE_NAME: &str = "deno-x86_64-apple-darwin.zip";
-#[cfg(target_os = "linux")]
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
 const ARCHIVE_NAME: &str = "deno-x86_64-unknown-linux-gnu.zip";
 
+#[cfg(any(
+  not(target_arch = "x86_64"),
+  target_os = "ios",
+  target_os = "android",
+  target_os = "freebsd",
+  target_os = "dragonfly",
+  target_os = "bitrig",
+  target_os = "openbsd",
+  target_os = "netbsd",
+))]
+const ARCHIVE_NAME: &str = NO_ARCHIVE;
```

Comment:
Maybe it would be better to use `if cfg!() { ... } else { ... } ` below  rather than specifying all of these targets (most of which we will likely never build against)

Reply:
Moved from `const` to function with `if cfg!() {} ...`